build/yamls/antrea.yml

```diff
@@ -264,6 +264,13 @@ data:
     # CIDR Range for services in cluster. It's required to support egress network policy, should
     # be set to the same value as the one specified by --service-cluster-ip-range for kube-apiserver.
     #serviceCIDR: 10.96.0.0/12
+
+    # Determines how pod traffic is encapsulated. It has the following options
+    # - encap (default)
+    # - noEncap
+    # - hybrid
+    # - noEncapMasq
+    #podTrafficEncapMode: encap
```

Comment:
Maybe add more details here about how each mode works, as this is user facing.
BTW, does `noEncapMasq` mean `passthrough` @jianjuns tried to add, I feel `noEncapMasq` might be a little confusing about whether it's going to masquerade or not and what's the masquerade behavior of `encap` and `noEncap`, if it indeed means `passthrough`, the later sounds more clear to me.

Reply:
yes, we can rename to passthrough/bump-in-wire,  if we are on the same page

Reply:
@jianjuns @antoninbas 
what do u think rename NoEncapMasq to 
1) passthrough or
2) BumpInWire

Thx, Su

Reply:
I am not really good at naming, but was thinking about passthrough or underlayNetwork. @antoninbas @abhiraut @salv-orlando might have some better ideas.

Reply:
Today @antoninbas, @abhiraut, and I chatted about this a little. Our thinking is that as we still need to close the design with public cloud (I thought we should be able to close it faster and then need to have the mode ready, but seems this is not the case), it is not very clear how this mode for public clouds will be exactly like. In this case, should we make it a separate PR and merge after we close the design? Otherwise seems hard to make sure we implement something really useful, or to discuss the naming. Let me know what you think.

Reply:
yes, noEncapMasq assumes there is underlaying CNI network already working (as in managed AKS, EKS), and Antrea is only do ingress/egress functionaries

Reply:
regarding to how this change relates to native public cloud support, please see design doc. It is mentioned a bit.
At this point, I am looking to use NoEncapMasq in conjunction with AKS and EKS CNI. That is AKS/EKS provide connectivity, Antrea provides policy